sensor/AdblockPlugin.js

```diff
@@ -275,14 +289,31 @@ class AdblockPlugin extends Sensor {
         writer.on('error', err => {
           reject(err);
         });
-        hashes.forEach((hash) => {
-          let line = util.format("hash-address=/%s/%s%s\n", hash.replace(/\//g, '.'), "", "$adblock")
+        if (fastMode) {
+          this.preprocess(key, hashes);
+          let line = util.format("redis-hash-match=/%s/%s%s\n", key, "", "$adblock");
           writer.write(line);
-        });
+        } else {
+          hashes.forEach((hash) => {
+            let line = util.format("hash-address=/%s/%s%s\n", hash.replace(/\//g, '.'), "", "$adblock")
+            writer.write(line);
+          });
+        }
         writer.end();
       });
     }
 
+    preprocess(key, hashes) {
+      rclient.del(key);
+      const newHashes = [];
+      hashes.forEach((hash) => {
+        newHashes.push(hash.replace(/\//g, '.'));
+      });
+      const cmd = [key];
+      cmd.push.apply(cmd, newHashes);
```

Comment:
Since there may be 10K+ entries, use concat instead of Array.prototype.push.apply to avoid call stack size exceeded
https://github.com/nodejs/node/issues/27732#issuecomment-493149690